crates/rooch-config/src/lib.rs

```diff
@@ -119,14 +119,14 @@ impl BaseConfig {
         let base_data_dir = match opt.base_data_dir.clone() {
             Some(base_data_dir) if base_data_dir.to_str() == Some("TMP") => temp_dir(),
             Some(base_data_dir) => DataDirPath::PathBuf(base_data_dir),
-            // only dev mode use temp store dir
-            None if chain_id.is_dev() => temp_dir(),
+            // only local network use temp store dir
+            None if chain_id.is_local() => temp_dir(),
```

Comment:
Do not use temp_dir as default. In the test case, we can use `TMP` as data dir to use temp_dir.

Reply:
default use `local` dir?

Reply:
If the default network is `local`, it is reasonable to use a temporary directory.

Reply:
`local` is for DApp developer. Most DApp developer wants to keep the data in developing progress. The framework or chain developer wants temp dir. 

DApp developer priority over chain developer.

Reply:
We can discuss this tomorrow.